spec/support/examples_group/shared_examples_for_ownership_mixin.rb

```diff
@@ -146,6 +146,63 @@
       end
     end
 
+    describe ".user_or_group_owned" do
+      let(:factory) { described_class.name == "Vm" ? :vm_vmware : described_class.table_name.singularize }
+      let(:owned_resource) { FactoryGirl.create(factory) }
+      let(:owning_user) {
+        FactoryGirl.create :user, :userid => "user_owner", :miq_groups => FactoryGirl.create_list(:miq_group, 1)
+      }
+
+      context "by user in this region" do
+        it "returns resource owned by user" do
+          owned_resource.evm_owner = owning_user
+          owned_resource.save!
+
+          expect(described_class.user_or_group_owned(owning_user, nil)).to eq([owned_resource])
+        end
+
+        it "returns resource owned by user or group" do
+          owned_resource.evm_owner = owning_user
+          owned_resource.save!
+
+          expect(described_class.user_or_group_owned(owning_user, owning_user.current_group)).to eq([owned_resource])
+        end
+      end
+
+      context "by user in a remote region" do
+        it "returns resource owned by user" do
+          my_region_number =  ApplicationRecord.my_region_number
+          remote_region_number = my_region_number + 1
+          remote_id = ApplicationRecord.region_to_range(remote_region_number).first
+
+          remote_owning_user = FactoryGirl.create :user, :id => remote_id, :userid => "user_owner"
+          owned_resource.id = remote_id
+          owned_resource.evm_owner = remote_owning_user
+          owned_resource.save!
+
+          expect(owned_resource.evm_owner_id).not_to eq(owning_user.id)
+          expect(described_class.user_or_group_owned(owning_user, nil)).to eq([owned_resource])
+        end
+
+        it "returns resource owned by user or group" do
+          my_region_number =  ApplicationRecord.my_region_number
+          remote_region_number = my_region_number + 1
+          remote_id = ApplicationRecord.region_to_range(remote_region_number).first
+
+          remote_owning_user = FactoryGirl.create :user, :id => remote_id, :userid => "user_owner", :miq_groups => FactoryGirl.create_list(:miq_group, 1)
```

Comment:
You could also make the `remote_owning_user` a `let` then assign the groups in this spec as that is what has actually changed.